pkg/openservicebroker/types.go

```diff
@@ -0,0 +1,157 @@
+	Name        string                 `json:"name"`
+	Description string                 `json:"description"`
+	Metadata    map[string]interface{} `json:"metadata,omitempty"`
+	Free        bool                   `json:"free,omitempty"`
+	Bindable    bool                   `json:"bindable,omitempty"`
+}
+
+type CatalogResponse struct {
+	Services []*Service `json:"services"`
+}
+
+type LastOperationResponse struct {
+	State       LastOperationState `json:"state"`
+	Description string             `json:"description,omitempty"`
+}
+
+type LastOperationState string
+
+const (
+	LastOperationStateInProgress LastOperationState = "in progress"
+	LastOperationStateSucceeded  LastOperationState = "succeeded"
+	LastOperationStateFailed     LastOperationState = "failed"
+)
+
+type ProvisionRequest struct {
+	ServiceID         string            `json:"service_id"`
+	PlanID            string            `json:"plan_id"`
+	Parameters        map[string]string `json:"parameters,omitempty"`
+	AcceptsIncomplete bool              `json:"accepts_incomplete,omitempty"`
+	OrganizationID    string            `json:"organization_guid"`
+	SpaceID           string            `json:"space_guid"`
+}
+
+type ProvisionResponse struct {
+	DashboardURL string    `json:"dashboard_url,omitempty"`
+	Operation    Operation `json:"operation,omitempty"`
+}
+
+type Operation string
+
+type UpdateRequest struct {
+	ServiceID         string            `json:"service_id"`
+	PlanID            string            `json:"plan_id,omitempty"`
+	Parameters        map[string]string `json:"parameters,omitempty"`
+	AcceptsIncomplete bool              `json:"accepts_incomplete,omitempty"`
+	PreviousValues    struct {
+		ServiceID      string `json:"service_id,omitempty"`
+		PlanID         string `json:"plan_id,omitempty"`
+		OrganizationID string `json:"organization_id,omitempty"`
+		SpaceID        string `json:"space_id,omitempty"`
+	} `json:"previous_values,omitempty"`
+}
+
+type UpdateResponse struct {
+	Operation Operation `json:"operation,omitempty"`
+}
+
+type BindRequest struct {
+	ServiceID    string `json:"service_id"`
+	PlanID       string `json:"plan_id"`
+	AppGUID      string `json:"app_guid,omitempty"`
+	BindResource struct {
+		AppGUID string `json:"app_guid,omitempty"`
+		Route   string `json:"route,omitempty"`
+	} `json:"bind_resource,omitempty"`
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+type BindResponse struct {
+	Credentials     map[string]interface{} `json:"credentials,omitempty"`
+	SyslogDrainURL  string                 `json:"syslog_drain_url,omitempty"`
+	RouteServiceURL string                 `json:"route_service_url,omitempty"`
+	VolumeMounts    []interface{}          `json:"volume_mounts,omitempty"`
+}
+
+type UnbindResponse struct {
+}
+
+type DeprovisionResponse struct {
+	Operation Operation `json:"operation,omitempty"`
+}
+
+type ErrorResponse struct {
+	Description string `json:"description"`
+}
+
+var AsyncRequired = struct {
```

Comment:
Indeed. The upstream spec specifies this specific combination of constants as a constant response.  The Error field does not appear in any other struct in the spec. `AsyncRequired` is used for one specific and important response. I wish Golang would permit this to be const, but it doesn't.  Given the above, I think var + anonymous struct is the correct definition to use.